test/test_integration.py

```diff
@@ -216,9 +216,306 @@ def test_upload_gallery(self):
 
         self.snapshot('test_upload_gallery')
 
+    def test_loading_component_initialization(self):
+        lock = Lock()
+
+        app = dash.Dash(__name__)
+
+        app.layout = html.Div([
+            dcc.Loading([
+                html.Div(id='div-1')
+            ], className='loading')
+        ], id='root')
+
+        @app.callback(
+            Output('div-1', 'children'),
+            [Input('root', 'n_clicks')]
+        )
+        def updateDiv(children):
+            with lock:
+                return 'content'
+
+        with lock:
```

Comment:
Using `with lock` instead of lock.acquire() / lock.release() as a test failure will throw an error and will prevent the lock from being released -- hence the test app will never terminate, blocking the tests indefinitely!

```
with lock:
    // Code
```

is equivalent to

```
try {
   lock.acquire()
    // Code
} finally {
    lock.release()
}